angr/exploration_techniques/tracer.py

```diff
@@ -195,7 +281,9 @@ def _update_state_tracking(self, state: 'angr.SimState'):
                 if self._compare_addr(self._trace[idx], addr):
                     idx += 1
                 else:
-                    raise Exception('BUG! Please investigate the claim in the comment above me')
```

Comment:
did you ever see any instances of this being tripped and it being an actual desync and not a unicorn bug?

Reply:
Missing, incomplete, or incorrect SimProcedure implementations.